src/libs/sys-util-rs/src/k8s.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+// Copyright (c) 2019-2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! Utilities to support K8S.
+//!
+//! This module depends on kubelet internal implementation details, a better way is needed
+//! to detect K8S EmptyDir medium type from `oci::spec::Mount` objects.
+
+use kata_types::mount;
+use oci::Spec;
+
+use crate::mount::get_linux_mount_info;
+
+pub use kata_types::k8s::is_empty_dir;
+
+/// Check whether the given path is a kubernetes ephemeral volume.
+///
+/// This method depends on a specific path used by k8s to detect if it's type of ephemeral.
+/// As of now, this is a very k8s specific solution that works but in future there should be a
+/// better way for this method to determine if the path is for ephemeral volume type.
```

Comment:
I don't believe this should go in this PR.  This is a dangerous optimization, so it needs more discussion and debate on its own, rather than in with all this other code.

Reply:
While agreed it isn't optimal, today we do key in on empty directories, configmaps as well as secrets, each of which have a very specific pattern on the filesystem when working with Kubelet.

Since this is contained in k8s.rs (good idea), and since this has been consistent with Kubernetes (and Kubernetes is our primary use case), I am okay with this.

We use this information today in the runtime to identify any volumes that would typically be used for fsnotify.

Reply:
Yeah, I hadn't realized we already did this.  I guess it makes sense to keep it then.  *barf*, though.